linker.ts

```diff
@@ -55,10 +65,12 @@ function replacePlaceholder (bytecode, label, address) {
  * @param libraries Mapping between fully qualified library names and the hex-encoded
  *     addresses they should be replaced with. Addresses shorter than 40 characters are automatically padded with zeros.
  *
+ * @param linkReferences A optional mapping of libraries to lists of placeholder positions in the binary.
+ *
  * @returns bytecode Hex-encoded bytecode string with placeholders replaced with addresses.
  *    Note that some placeholders may remain in the bytecode if `libraries` does not provide addresses for all of them.
  */
-function linkBytecode (bytecode: string, libraries: LibraryAddresses): string {
+function linkBytecode (bytecode: string, libraries: LibraryAddresses, linkReferences?: LinkReferences): string {
   assert(typeof bytecode === 'string');
   assert(typeof libraries === 'object');
 
```

Comment:
Since other param types are being asserted, should we do the same with `linkReferences`?
```
assert(!linkReferences || typeof linkReferences === 'object');
```
Not sure if that makes sense in this case...

Reply:
I think this is unnecessary here for the `linkReferences`, since it is an optional parameter, and it is only passed to `replacePlaceholder` where its existence is checked.